rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -68,9 +69,17 @@ const std::chrono::milliseconds
 Player::queue_read_wait_period_ = std::chrono::milliseconds(100);
 
 Player::Player(
-  std::shared_ptr<rosbag2_cpp::Reader> reader, std::shared_ptr<Rosbag2Node> rosbag2_transport)
-: reader_(std::move(reader)), rosbag2_transport_(rosbag2_transport)
-{}
+  std::shared_ptr<rosbag2_cpp::Reader> reader,
+  std::shared_ptr<Rosbag2Node> rosbag2_transport,
+  const rosbag2_storage::BagMetadata & metadata)
+: reader_(std::move(reader)), rosbag2_transport_(rosbag2_transport), metadata_(metadata)
+{
+  for (auto topic_information : metadata.topics_with_message_count) {
+    const auto & topic_metadata = topic_information.topic_metadata;
+    const auto & loaded_profiles = YAML::Load(topic_metadata.offered_qos_profiles);
+    recorded_qos_profiles_[topic_metadata.name] = loaded_profiles.as<std::vector<Rosbag2QoS>>();
```

Comment:
What about the unhappy path? I.e. YAML is not correct. Did we test this? Are we printing out a reasonable error message?

I would also suggest to update the documentation to explicitly state what type of exception may get thrown from the ctor. Seems like we're moving from sonething that never could fail, to something that can "easily" fail (given bad input). It's worth checking the implication on the codebase.